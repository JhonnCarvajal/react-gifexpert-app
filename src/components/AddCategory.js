import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e) => {
    //console.log(e.target.value);
    setInputValue(e.target.value);
    //console.log("Handle input Change llamado");
  };
  const hanldeSubmit = (e) => {
    e.preventDefault();
    //console.log("Handle submit",inputValue);
    if (inputValue.trim().length > 1) {
      setCategories((cats) => [inputValue, ...cats]); //se paso como props una funcion(setCategories) aqui se puede ver la utilidad
      // de usar el callback del use State para acceder al estado anterior, pues no contamos con ese estado directamente, al ser otro componente
      setInputValue("");
    }
  };
  return (
    <form onSubmit={hanldeSubmit}>
      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
