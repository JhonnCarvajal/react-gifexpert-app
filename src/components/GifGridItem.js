import React from "react";

export const GifGridItem = (props) => {
  //console.log(props);
  return <div className={'card animate__animated animate__fadeIn'}>
      {props.title}
      <img src={props.url} alt = 'Cargando...'/>
      </div>;
};
