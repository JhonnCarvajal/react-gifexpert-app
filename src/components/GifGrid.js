import React from "react";
import { useFetcGifs } from "../hook/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";
export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetcGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid animate__animated animate__fadeIn">
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
