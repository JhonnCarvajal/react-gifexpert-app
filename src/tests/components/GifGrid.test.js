import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetcGifs } from "../../hook/useFetchGifs";
import '@testing-library/jest-dom';
jest.mock("../../hook/useFetchGifs");
describe("Pruebas en GiFgRID", () => {
  const category = "IT";

  test("Debe crear y comparar snapshot", () => {
      //Para saimular la respuesta de una funcion(mock) se debe instanciar el objeto(wrapper) despues
      useFetcGifs.mockReturnValue({
          data:[],
          images:true
      });
      const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar imagenes cuando se cargan", () => {
      const img = [{
          id:'ABC',
          url:'https://google.com.jpg',
          title:'Cualquier titulo'
      },
      {
        id:'ABC123',
        url:'https://google.com.jpg',
        title:'Cualquier titulo'
    }]
    useFetcGifs.mockReturnValue({
        data:img,
        images:true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    const p = wrapper.find('p');
    expect(p.exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(img.length)
  });
});
