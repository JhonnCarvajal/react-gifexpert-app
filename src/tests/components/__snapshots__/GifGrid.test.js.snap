// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiFgRID Debe crear y comparar snapshot 1`] = `
<Fragment>
  <h3>
    IT
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Pruebas en GiFgRID Debe de mostrar imagenes cuando se cargan 1`] = `
<Fragment>
  <h3>
    IT
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier titulo"
      url="https://google.com.jpg"
    />
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="Cualquier titulo"
      url="https://google.com.jpg"
    />
  </div>
</Fragment>
`;
